draft-ietf-quic-version-negotiation.md

```diff
@@ -225,42 +282,49 @@ received Version Negotiation packet, so clients MUST NOT add or remove reserved
 version from that list.
 
 
-# Version Downgrade Prevention
+# Version Downgrade Prevention {#downgrade}
 
 Clients MUST ignore any received Version Negotiation packets that contain the
 version that they initially attempted. Once a client has reacted to a Version
 Negotiation packet, it MUST drop all subsequent Version Negotiation packets.
 
 Servers MUST validate that the client's `Currently Attempted Version` matches
-the version in the long header that carried the transport parameter. Similarly,
-clients MUST validate that the server's `Negotiated Version` matches the long
-header version. If an endpoint's validation fails, it MUST close the connection
-with an error of type VERSION_NEGOTIATION_ERROR.
-
-When a server parses the client's `version_negotiation` transport parameter, if
-the `Received Negotiation Version Count` is not zero, the server MUST validate
-that it could have sent the Version Negotation packet described by the client
-in response to an Initial of version `Previously Attempted Version`.
-In particular, the server MUST ensure that there are no versions that it
-supports that are absent from the Received Negotiation Versions, and that the
-ordering matches the server's preference. If this validation fails, the server
-MUST close the connection with an error of type VERSION_NEGOTIATION_ERROR. This
+the version in the long header that carried the handshake version information.
+Similarly, clients MUST validate that the server's `Negotiated Version` matches
+the long header version. If an endpoint's validation fails, it MUST close the
+connection. If the connection was using QUIC version 1, it MUST be closed with
+a transport error of type `VERSION_NEGOTIATION_ERROR`.
+
+When a server parses the client's handshake version information, if the
+`Received Negotiation Version Count` is not zero, the server MUST validate that
+it could have sent the Version Negotation packet described by the client in
```

Comment:
could? Perhaps would?

Reply:
Not really, "would" implies that the server doesn't use randomness when generating VN packets - and we do that for GREASE